Gemfile

```diff
@@ -30,6 +30,9 @@ gem "bootstrap_form", ">= 4.2.0"
 
 group :development, :test do
   gem "byebug", platforms: [:mri, :mingw, :x64_mingw]
+  gem "rspec-rails"
+  gem "capybara", ">= 2.15"
+  gem "factory_bot_rails"
```

Comment:
gem này dùng làm gì?

Reply:
- gem "rspec-rails" : là gem dùng để kiểm thử các chức năng trong rails, cho phép viết unit tests cho controller, views, helpers và model trong ứng dụng Rails.
Cách hoạt động : 
Rspec sử dụng những từ như "describe" và "it" để thể hiện hành vi, tổ chức của codes . Describe dùng để nhóm các test case theo class, module, method hoặc theo action của controller, ví dụ : 
RSpec.describe "something" do
   it "does something" do
   end
 end
Khi sử dụng `describe` , cần lưu ý với class, class method và instance method khi mô tả:
- describe "ClasName" do end;
- describe "#instance_method" do end;
- describe ".class_method" do end;
`context`: dùng để nhóm các điều kiện của example
`it`: được sử dụng để định nghĩa một "Ví dụ" hay nói cách khác là dùng để định nghĩa 1 test case
`expect`: được sử dụng để xác định "Kỳ vọng" của 1 case trong RSpec.
let và let!: dùng để tạo data trong example